mitodl/mit-open
# navigation menu

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4179

### Description (What does it do?)
This PR adds the `NavDrawer` reusable component to `ol-components` and implements it site-wide in the header via `frontends/mit-open/src/page-components/Header/Header.tsx`. Various links were added that point to the search page with various parameters applied for quick access to filtered searches.

### Screenshots (if appropriate):
<img width="1919" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/ac1827c3-ed6e-41f3-934e-142836792acf">
<img width="289" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/3c0c59c4-7f01-44a5-be7e-096db5bf45f5">

### How can this be tested?
 - Simply spin up `mit-open` on this branch and visit http://localhost:8063
 - Verify that the nav drawer loads and that the links go where they say they should
 - Test the open / close functionality of the drawer under a variety of conditions:
   - Desktop
     - Clicking the "Explore MIT" button in the header should open the drawer if it's closed and close the drawer if it's open
     - Clicking anywhere outside of the drawer while it is open should close it
     - If you are on the search page and a resource drawer is open, clicking the "Explore MIT" button to open the nav drawer should close the resource drawer before opening the nav drawer
   - Mobile
     - All of the above conditions should also apply
